File: Data Structures & Algorithms/combination-target-sum/submission-0.cpp
class Solution {
public:
    vector<vector<int>>res;
    void helper(int ind,vector<int>& nums,vector<int>&ans,int target){
       if(target==0){
        res.push_back(ans);
        return ;
       }
       if(ind==nums.size()){
        
        return;
       }

       
    //    take
       if(target>=nums[ind]){
       ans.push_back(nums[ind]);

       helper(ind,nums,ans,target-nums[ind]);
       ans.pop_back();

       }


       helper(ind+1,nums,ans,target);



    }
   

    vector<vector<int>> combinationSum(vector<int>& nums, int target) {
    vector<int>ans;
    helper(0,nums,ans,target);

    return res;


    }
};

File: Data Structures & Algorithms/check-if-array-is-sorted-and-rotated/submission-0.java
class Solution {
    public boolean check(int[] nums) {
      int n=nums.length;
      int cnt=0;
      for(int i=0;i<nums.length-1;i++){
        if(nums[i]>nums[i+1]){
           cnt++;
        }
      }
      if(nums[n-1]>nums[0]) cnt++;
      return (cnt>1)?false:true;
        
    }
}